Source/WebCore/page/Quirks.cpp

```diff
@@ -2160,1043 +2436,988 @@ void Quirks::setTopDocumentURLForTesting(URL&& url)
+static void handleScriptToEvaluateBeforeRunningScriptFromURLQuirk(QuirksData& quirksData, const URL& /* quirksURL */, const String& topDomain, const URL& /* documentURL */)
 {
-    UNUSED_PARAM(quirksURL);
-    UNUSED_PARAM(documentURL);
-
     if (topDomain == "thesaurus.com"_s) {
         quirksData.isThesaurus = true;
-        quirksData.needsScriptToEvaluateBeforeRunningScriptFromURLQuirk = true;
+        quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsScriptToEvaluateBeforeRunningScriptFromURLQuirk);
     }
 
 #if ENABLE(DESKTOP_CONTENT_MODE_QUIRKS)
     if (topDomain == "webex.com"_s) {
         quirksData.isWebEx = true;
-        quirksData.needsScriptToEvaluateBeforeRunningScriptFromURLQuirk = true;
+        quirksData.enableQuirk(QuirksData::SiteSpecificQuirk::NeedsScriptToEvaluateBeforeRunningScriptFromURLQuirk);
     }
 #endif
 }
 #endif
 
 #if PLATFORM(IOS_FAMILY) || PLATFORM(MAC)
-static void handleICloudQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleICloudQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& /* documentURL */)
 {
-    if (quirksDomainString != "icloud.com"_s)
+    if (quirksDomainString != "icloud.com"_s) [[unlikely]]
         return;
 
-    UNUSED_PARAM(documentURL);
 #if PLATFORM(IOS_FAMILY)
     // icloud.com rdar://131836301
-    quirksData.shouldSilenceWindowResizeEventsDuringApplicationSnapshotting = quirksURL.path().contains("mail"_s) || quirksURL.fragmentIdentifier().contains("mail"_s);
+    bool shouldSilenceWindowResizeEventsDuringApplicationSnapshotting = quirksURL.path().contains("mail"_s) || quirksURL.fragmentIdentifier().contains("mail"_s);
+    quirksData.setQuirkState(QuirksData::SiteSpecificQuirk::ShouldSilenceWindowResizeEventsDuringApplicationSnapshotting, shouldSilenceWindowResizeEventsDuringApplicationSnapshotting);
 #endif
 #if PLATFORM(MAC)
     // icloud.com rdar://26013388
-    quirksData.isNeverRichlyEditableForTouchBarQuirk = quirksURL.path().contains("notes"_s) || quirksURL.fragmentIdentifier().contains("notes"_s);
+    bool isNeverRichlyEditableForTouchBarQuirk = quirksURL.path().contains("notes"_s) || quirksURL.fragmentIdentifier().contains("notes"_s);
+    quirksData.setQuirkState(QuirksData::SiteSpecificQuirk::IsNeverRichlyEditableForTouchBarQuirk, isNeverRichlyEditableForTouchBarQuirk);
 #endif
 }
 #endif
 
-static void handleTMobileQuirks(QuirksData& quirksData, const URL& quirksURL, const String& quirksDomainString, const URL& documentURL)
+static void handleScribdQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL& /* documentURL */)
+{
+    if (quirksDomainString != "scribd.com"_s) [[unlikely]]
+        return;
+
+    // Remove this once rdar://142573562 is resolved.
```

Comment:
Please verify this is the correct radar.

The comments in the quirks show it is also being used for steam too.

Reply:
Good catch! This was an old copy/paste error. I will fix it. Reviewing that also revealed that an old `std::optional<bool>` was present we don't need anymore.